### PR TITLE
Add mute timer wallet

### PR DESCRIPTION
**Pull request by winterweixiaodong**
1. add mutex for wallet manager
2. add a timer for wallet unlock, if a password is input, there is no need to input the  password within 5 mins

### CODE REVIEW

File: logic/logic.go [hunk 1]
```diff
@@ -142,7 +219,7 @@ func GetBalance(address core.Address, db storage.Storage) (*common.Amount, error
 
 //get all addresses
 func GetAllAddressesFromTest() ([]core.Address, error) {
```
Comment (yeeshi) [inline logic/logic.go]: Testing function should be refactored and moved to testing purpose files

File: rpc/rpc_service.go [hunk 2]
```diff
@@ -47,31 +47,68 @@ func (rpcSerivce *RpcService) RpcCreateWallet(ctx context.Context, in *rpcpb.Cre
...
-			addr = ""
+			} else {
+				msg = "Create Wallet Error: Wallet Empty!"
+				addr = ""
+			}
+		} else { //unlock
+			wallet, err := logic.AddWallet()
+			if err != nil {
+				msg = err.Error()
+			} else if wallet != nil {
```
Comment (yeeshi) [inline rpc/rpc_service.go]: This seems to be logic. Is it better to put them in logic package?

File: logic/logic.go (hunk 1 above)
Comment (yeeshi) [inline logic/logic.go]: This testing purpose function should be refactored/moved to testing related files

Comment (winterweixiaodong) [reply logic/logic.go]: changed

**Pull request merged**